### PR TITLE
Lazy Mint Overview

### PR DESCRIPTION
**Pull request by kolberszymon**

### CODE REVIEW

File: docs/overview/lazyMintOverview.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Introduction
+
+In order to be able to fully understand what lazy minting is, and the benefits of doing so, we have to scratch the other aspects of blockchain and NFTs first. This document will touch three main aspects which in my opinion are crucial in lazy minting process understanding:
+
+1. Technical description of NFT,
+2. Minting
+3. Lazy Minting
+
+### Technical description of NFT
```
Comment (mazinden) [inline docs/overview/lazyMintOverview.md:14]: It seems that the description is more suitable for “How to prepare for the NFT minting”
Maybe, we can change heading or restructure text

File: docs/overview/lazyMintOverview.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+2. Invoke Smart Contract function responsible for minting, and assign URI created in the earlier step to your account address. That's it. Since code on the blockchain is immutable, and we have tokenId variable which tracks every unique NFT creation on a given smart contract, even if the same URI would be used twice, it will still be a different NFT.
+
+### Minting
+
+If you've read the first point, this one will be a repetition. Minting can be described as a process of a write operation (it's important because if you're writing on the blockchain, you're paying a gas fee), which has one goal. Associate address of a creator with a tokenId, which points to token metadata URI on IPFS. Pardon those who thought that it is more complicated. Of course, there are some language-specific restrictions and security steps that we have to take care of, but it's enough to use OpenZeppelin implementation of ERC721/ERC1155 to safely elevate core functionality, which in this case is NFT creation.
```
[the remark this answers is not in the document]
Comment (kolberszymon) [reply docs/overview/lazyMintOverview.md]: I think the gas is self-explanatory, i.e. if I'm correct on every blockchain you have some sort of fee, that you need to pay for a transaction, and this doc is about lazy minting functionality

[the remark this answers is not in the document]
Comment (kolberszymon) [reply docs/overview/lazyMintOverview.md]: Which is blockchain independent

**Review by VitalyLyaper — COMMENTED** [review]
on a high level following is missing - what events we are listening, what object we are creating of them
[6:19](https://rarible.slack.com/archives/D037Q6CMH5Y/p1648048745259359)
it is not described that we have per blockchain listeners and on a multichain level we make abstraction
[6:19](https://rarible.slack.com/archives/D037Q6CMH5Y/p1648048777562289)
also question for re-indexing is not covered - after blockchain restructuring
[6:22](https://rarible.slack.com/archives/D037Q6CMH5Y/p1648048923694369)
according to our architecture we also notifying marketplaces simultaneously when indexing done, without real fetching from DB

**Pull request merged**